### ISSUE TITLE
[error][Error] Error: Call to undefined function nystudio107\imageoptimize\lib\imagecreatefromjpeg() in /var/www/vendor/nystudio107/craft-imageoptimize/src/lib/Potracio.php:194

### ISSUE THREAD

**Issue by Saboteur777** [nystudio107/craft3-imageoptimize#44]
```
2018-02-09 23:12:05 [172.22.0.1][1][-][info][nystudio107\imageoptimize\services\OptimizedImages::addVariantImageToModel] URL created: /images/articles/_1200x743_crop_center-center_82_line/phil-botha-424950.jpg
2018-02-09 23:12:06 [172.22.0.1][1][-][error][craft\queue\QueueLogBehavior::afterError]  [244] Resaving Assets in Cikk képek (attempt: 1) - Error (time: 0.320s): Call to undefined function nystudio107\imageoptimize\lib\imagecreatefromjpeg()
2018-02-09 23:12:06 [172.22.0.1][1][-][error][Error] Error: Call to undefined function nystudio107\imageoptimize\lib\imagecreatefromjpeg() in /var/www/vendor/nystudio107/craft-imageoptimize/src/lib/Potracio.php:194
Stack trace:
#0 /var/www/vendor/nystudio107/craft-imageoptimize/src/services/Placeholder.php(130): nystudio107\imageoptimize\lib\Potracio->loadImageFromFile('/var/www/storag...')
#1 /var/www/vendor/nystudio107/craft-imageoptimize/src/services/OptimizedImages.php(171): nystudio107\imageoptimize\services\Placeholder->generatePlaceholderSvg('/var/www/storag...')
#2 /var/www/vendor/nystudio107/craft-imageoptimize/src/services/OptimizedImages.php(249): nystudio107\imageoptimize\services\OptimizedImages->generatePlaceholders(Object(craft\elements\Asset), Object(nystudio107\imageoptimize\models\OptimizedImage), 1.6137472283814)
#3 /var/www/vendor/nystudio107/craft-imageoptimize/src/services/OptimizedImages.php(101): nystudio107\imageoptimize\services\OptimizedImages->addVariantImageToModel(Object(craft\elements\Asset), Object(nystudio107\imageoptimize\models\OptimizedImage), Object(craft\models\AssetTransform), Array, 1.6137472283814)
#4 /var/www/vendor/nystudio107/craft-imageoptimize/src/services/OptimizedImages.php(274): nystudio107\imageoptimize\services\OptimizedImages->populateOptimizedImageModel(Object(craft\elements\Asset), Array, Object(nystudio107\imageoptimize\models\OptimizedImage))
#5 /var/www/vendor/nystudio107/craft-imageoptimize/src/jobs/ResaveOptimizedImages.php(77): nystudio107\imageoptimize\services\OptimizedImages->updateOptimizedImageFieldData(Object(nystudio107\imageoptimize\fields\OptimizedImages), Object(craft\elements\Asset))
#6 /var/www/vendor/yiisoft/yii2-queue/src/Queue.php(213): nystudio107\imageoptimize\jobs\ResaveOptimizedImages->execute(Object(craft\queue\Queue))
#7 /var/www/vendor/yiisoft/yii2-queue/src/cli/Queue.php(139): yii\queue\Queue->handleMessage('244', 'O:52:"nystudio1...', '300', 1)
#8 /var/www/vendor/craftcms/cms/src/queue/Queue.php(95): yii\queue\cli\Queue->handleMessage('244', 'O:52:"nystudio1...', '300', 1)
#9 /var/www/vendor/craftcms/cms/src/controllers/QueueController.php(84): craft\queue\Queue->run()
#10 /var/www/vendor/craftcms/cms/src/controllers/QueueController.php(104): craft\controllers\QueueController->actionRun()
#11 [internal function]: craft\controllers\QueueController->actionRetry()
#12 /var/www/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#13 /var/www/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#14 /var/www/vendor/craftcms/cms/src/web/Controller.php(80): yii\base\Controller->runAction('retry', Array)
#15 /var/www/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('retry', Array)
#16 /var/www/vendor/craftcms/cms/src/web/Application.php(241): yii\base\Module->runAction('queue/retry', Array)
#17 /var/www/vendor/craftcms/cms/src/web/Application.php(451): craft\web\Application->runAction('queue/retry', Array)
#18 /var/www/vendor/craftcms/cms/src/web/Application.php(212): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#19 /var/www/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#20 /var/www/html/index.php(21): yii\base\Application->run()
#21 {main}
2018-02-09 23:12:06 [172.22.0.1][1][-][info][yii\db\Command::execute] UPDATE `queue` SET `fail`=1, `dateFailed`='2018-02-09 22:12:06', `error`='Call to undefined function nystudio107\\imageoptimize\\lib\\imagecreatefromjpeg()' WHERE `id`='244'
2018-02-09 23:12:06 [172.22.0.1][1][-][profile begin][yii\db\Command::execute] UPDATE `queue` SET `fail`=1, `dateFailed`='2018-02-09 22:12:06', `error`='Call to undefined function nystudio107\\imageoptimize\\lib\\imagecreatefromjpeg()' WHERE `id`='244'
2018-02-09 23:12:06 [172.22.0.1][1][-][profile end][yii\db\Command::execute] UPDATE `queue` SET `fail`=1, `dateFailed`='2018-02-09 22:12:06', `error`='Call to undefined function nystudio107\\imageoptimize\\lib\\imagecreatefromjpeg()' WHERE `id`='244'
2018-02-09 23:12:06 [172.22.0.1][1][-][info][yii\db\Command::execute] UPDATE `queue` SET `dateReserved`=NULL, `timeUpdated`=NULL, `progress`=0 WHERE `timeUpdated` < 1518214326 - `ttr`
2018-02-09 23:12:06 [172.22.0.1][1][-][profile begin][yii\db\Command::execute] UPDATE `queue` SET `dateReserved`=NULL, `timeUpdated`=NULL, `progress`=0 WHERE `timeUpdated` < 1518214326 - `ttr`
2018-02-09 23:12:06 [172.22.0.1][1][-][profile end][yii\db\Command::execute] UPDATE `queue` SET `dateReserved`=NULL, `timeUpdated`=NULL, `progress`=0 WHERE `timeUpdated` < 1518214326 - `ttr`
2018-02-09 23:12:06 [172.22.0.1][1][-][info][yii\db\Command::query] SELECT *
FROM `queue`
WHERE ((`fail`=FALSE) AND (`timeUpdated` IS NULL)) AND (`timePushed` <= 1518214326 - `delay`)
ORDER BY `priority`, `id`
LIMIT 1
2018-02-09 23:12:06 [172.22.0.1][1][-][profile begin][yii\db\Command::query] SELECT *
FROM `queue`
WHERE ((`fail`=FALSE) AND (`timeUpdated` IS NULL)) AND (`timePushed` <= 1518214326 - `delay`)
ORDER BY `priority`, `id`
LIMIT 1
2018-02-09 23:12:06 [172.22.0.1][1][-][profile end][yii\db\Command::query] SELECT *
FROM `queue`
WHERE ((`fail`=FALSE) AND (`timeUpdated` IS NULL)) AND (`timePushed` <= 1518214326 - `delay`)
ORDER BY `priority`, `id`
LIMIT 1

```

**Comment by khalwat** [comment]
It's the same as this issue:

https://github.com/nystudio107/craft-imageoptimize/issues/42

You don't have GD installed as part of your PHP installation.

**Comment by khalwat** [comment]
Addressed in 1.4.14; it will no longer attempt to generate a placeholder silhouette if you do not have `GD` installed, and it will give you a warning message on the Settings page to that effect:

https://github.com/nystudio107/craft-imageoptimize/releases/tag/1.4.14

**Comment by Saboteur777** [comment]
Thank you!
